kong/plugins/pre-function/_handler.lua

```diff
@@ -14,14 +14,27 @@ return function(plugin_name, priority)
   function ServerlessFunction:access(config)
 
     local functions = config_cache[config]
+
     if not functions then
+      -- first call, go compile the functions
       functions = {}
       for _, fn_str in ipairs(config.functions) do
-        local func1 = loadstring(fn_str)
-        local _, func2 = pcall(func1)
-        insert(functions, type(func2) == "function" and func2 or func1)
+        local func1 = loadstring(fn_str)    -- load it
+        local _, func2 = pcall(func1)       -- run it
```

Comment:
While I think this is a step in the right direction, I'm wondering if immediate invocation could have unforeseen side-effects with old functions.

Reply:
this is in "access" phase, so there is no difference? first request, it compiles and runs, and gets cached. 2nd and later, it runs from cache...

Reply:
I see, I didn't look at the phase when reviewing this portion of the code, that reads correct and wouldn't cause side-effects. Ran a few tests as well, looks good on my side.